app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -251,6 +292,25 @@ class AgendaViewModel @Inject constructor(
                 }
                 logout()
             }
+            is AgendaEvent.OneTimeEvent -> {
+                when (event) {
+                    is AgendaEvent.OneTimeEvent.Reset -> {
+                        // Reset the one time event
+                        _agendaState.update {
+                            it.copy(oneTimeEvent = null)
+                        }
+                        yield()
+                    }
+                    else -> {
+                        // Send the one time event
+                        yield()
+                        _agendaState.update {
+                            it.copy(oneTimeEvent = event)
+                        }
+                        yield()
```

Comment:
I'm actually curious why you use all these `yield()` calls across your project, I have never used that in my life and was always good 🤔 😂 
What would break if you would remove these calls?

Reply:
I want to be sure the "AgendaItems" are updated properly, otherwise I get race conditions. 

Is there another way to update `AgendaItems` and have it processed by the `stateIn` fully before proceeding to the next step? Otherwise the "intermediate" states could be lost or ignored. The `stateIn` function needs time to run on the "looper". Without using the `yield()`, I was getting race conditions.

Can you suggest another way of structuring my code in a way that will respect the `stateIn` processing and allow events to be seen and not ignored?

Reply:
Alright, usually `stateFlow.update` should take care of all race conditions, but if it works for you I'm not saying anything against that